docs/multiple-canaries-guide.md

```diff
@@ -0,0 +1,126 @@
+
+# Multiple Canaries Guide
+
+- [Multiple Canaries Guide](#multiple-canaries-guide)
+  - [Background](#background)
+  - [Local Canary](#local-canary)
+  - [Global Canary](#global-canary)
+  - [Practical Guide](#practical-guide)
+    - [Explicitly Exclusive Traffic Rules](#explicitly-exclusive-traffic-rules)
+    - [Choose Zero or One Canary](#choose-zero-or-one-canary)
+  - [FAQ](#faq)
+
+## Background
+
+Canary deployment normally aims to test new features of services with a specific part of the traffic in the production environment. And the feature will be evaluated in multi-dimensions such as online errors, performance or business feedback from users, etc.
+
+In the concept, the canary sounds simple, but in the real world, we often need to handle more complicated things, for example:
+
+- Test and choose one decision from multiple candidates of new features, which means that multiple canaries are running for one service technically.
+- Simultaneously, a service has multiple canaries testing for different features.
+- The situations above expanded across multiple services.
+
+In a simple summary, we need to explicitly handle how to run multiple canaries in one or more services, and constitute the relationship among them.
+
+## Local Canary
+
+First of all, we start with the simplest situation named `local canary`: If a feature only needs one service to deploy a canary, then we call this canary `local canary`. Please notice there could be many `local canary` at the same time, even within one service. So we could make a definition: `local canary` is the feature only requiring one service to deploy canary.
+
+The local canary itself is very simple, but when it comes to the relationship between them we need to be careful. So we will use an evolutionary way to illustrate its points.
+
+As the basic example at (1), we have 3 services to represent backend services of an order takeaway app, and primary traffic means all traffic except canary traffic:
+
+![image](imgs/multiple-canaries-guide-01.png)
+
+Then at (2), `team-delivery` deployed a local canary `delivery-beijing` to test a new feature for traffic from Beijing.
```

Comment:
Propose the use "the delivery team" instead of "team-delivery" (also "the restaurant team"), it is a little confusing when used together with "delivery-beijing", "restaurant-beijing".

Reply:
Updated the description, but the ones in diagrams are not modified cause they are not confusing.